ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -181,32 +183,71 @@ azure_add_reference(WT_STORAGE_SOURCE *storage_source)
     return 0;
 }
 
+// Flush a file locally to Azure.
 static int
 azure_flush(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session, WT_FILE_SYSTEM *file_system,
   const char *source, const char *object, const char *config)
 {
     WT_UNUSED(storage_source);
-    WT_UNUSED(session);
-    WT_UNUSED(file_system);
-    WT_UNUSED(source);
-    WT_UNUSED(object);
     WT_UNUSED(config);
+    azure_file_system *azure_fs = reinterpret_cast<azure_file_system *>(file_system);
+    WT_FILE_SYSTEM *wtFileSystem = azure_fs->wt_fs;
 
-    return 0;
+    if (!std::filesystem::exists(object)) {
+        std::cerr << "azure_flush: Object: " << object << " does not exist." << std::endl;
+        return ENOENT;
+    }
+
+    bool exists_native = false;
+    int ret = wtFileSystem->fs_exist(
+      wtFileSystem, session, std::filesystem::canonical(source).string().c_str(), &exists_native);
+    if (ret != 0) {
+        std::cerr << "azure_flush: Failed to check for the existence of " << source
+                  << " on the native filesystem.";
+        return ret;
+    }
+
+    if (exists_native == false) {
+        std::cerr << "azure_flush: " << source << " No such file.";
+        return ENOENT;
+    }
+
+    std::cout << "azure_flush: Uploading object: " << object << " into bucket using put_object."
+              << std::endl;
+
+    // Upload the object into the bucket.
+    if (azure_fs->azure_conn->put_object(object, std::filesystem::canonical(source)) != 0)
+        std::cerr << "azure_flush: Put object request to Azure failed." << std::endl;
+    else
+        std::cout << "azure_flush: Uploaded object to Azure." << std::endl;
+    return ret;
 }
 
+// Check that flush has been completed by checking the object exists in Azure.
 static int
 azure_flush_finish(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session,
   WT_FILE_SYSTEM *file_system, const char *source, const char *object, const char *config)
 {
     WT_UNUSED(storage_source);
     WT_UNUSED(session);
-    WT_UNUSED(file_system);
-    WT_UNUSED(source);
-    WT_UNUSED(object);
     WT_UNUSED(config);
+    WT_UNUSED(source);
+    azure_file_system *azure_fs = reinterpret_cast<azure_file_system *>(file_system);
 
-    return 0;
+    std::cout << "azure_flush_finish: Checking object: " << object << " exists in Azure."
+              << std::endl;
+
+    // Check whether the object exists in the cloud.
+    bool exists_cloud = false;
+    azure_fs->azure_conn->object_exists(object, exists_cloud);
+    if (exists_cloud == true) {
```

Comment:
ditto. I would pefer check existence of it to be false then return, and then remove the else case and just turn it into a message. What do you think?

Reply:
`if (!exists_cloud) {
        std::cerr << "azure_flush_finish: Object: " << object << " does not exist in Azure."
                  << std::endl;
        return ENOENT;
    }
    std::cout << "azure_flush_finish: Object: " << object << " exists in Azure." << std::endl;
    return 0;
  `
  
  not too sure if this is what you're looking for, but i've done the check and removed the else case since its accounted for

Reply:
yeah was looking to do something similar here

Reply:
resolved.